receiver/otlpreceiver/otlphttp.go

```diff
@@ -28,3 +35,33 @@ type xProtobufMarshaler struct {
 func (*xProtobufMarshaler) ContentType() string {
 	return "application/x-protobuf"
 }
+
+var jsonMarshaller = &jsonpb.Marshaler{}
+
+// OTLPErrorHandler encodes the HTTP error message inside a rpc.Status message as required
+// by the OTLP protocol.
+func OTLPErrorHandler(w http.ResponseWriter, r *http.Request, err error, statusCode int) {
+	var msg []byte
+	var s *status.Status
+
+	switch statusCode {
+	case http.StatusBadRequest:
+		s = status.New(codes.InvalidArgument, err.Error())
+	default:
+		s = status.New(codes.Internal, err.Error())
+	}
+
+	contentType := r.Header.Get("Content-Type")
+	switch contentType {
+	case "application/json":
+		buf := new(bytes.Buffer)
+		jsonMarshaller.Marshal(buf, s.Proto())
```

Comment:
I went through the different inputs for this function and couldn't figure out a scenario when this would error out (as err is essentially a string input), so I have removed the fallback handling. 
Please let me know if I need to add that back again.